### ISSUE TITLE
No examples of how to use use restangularizeElement

### ISSUE THREAD

**Issue by WillsB3** [mgonto/restangular#953]
Hi,

I'm using Restangular in a project where I have a Tag resource, and each Tag resource has an array of TagInstances. I think I need to manually Restangularize the TagInstances once the Tag response comes back, but there are no examples of how to use this method in the documentation.

Could we get an/some examples added?

**Comment by grabbou** [comment]
To add restangularizeCollection with array examples as well.

**Comment by NickolasAcosta** [comment]
+1

**Comment by borisano** [comment]
+1. Would be great to see some documentation

**Comment by aug-riedinger** [comment]
:+1:

**Comment by mohsinkerai** [comment]
+1

**Comment by rguruprakash** [comment]
+1

**Comment by bitflower** [comment]
Would especially love to see the first parameter explained.

**Comment by daviesgeek** [comment]
@WillsB3 There is [a section](https://github.com/mgonto/restangular#restangular-methods) that documents `restangularizeElement`. Is there something else that needs clarifying? Other than the first parameter (see next paragraph), it's pretty clear?

I do agree with [@bitflower](https://github.com/mgonto/restangular/issues/953#issuecomment-218480922); it would be nice to have documentation on the first parameter

**Comment by wayspurrchen** [comment]
+1. The method is in the source code here (link is latest commit on master): https://github.com/mgonto/restangular/blob/b21e357ae6f4092d12cda2f6456f299d90b1e922/dist/restangular.js#L956

**Comment by daviesgeek** [comment]
That is correct. I wouldn't mind seeing a PR to explain this more in depth in the documentation

**Comment by thinkingmedia** [comment]
I'll post this here, because this is where Google is sending people looking for help.

```
restangularizeElement(parent, element, route, queryParams): Restangularizes a new element
```

"parent" is a Restangular object or `null` if there is no parent. Given the REST route `/accounts/123/buildings/123` the `buildings` object has `accounts` as a Restangular parent object. So if you had to serialize the `buildings` object and wanted to maintain the parent you would pass the `accounts` instance as the parent.

"element" is a JavaScript object with properties that you want to be a Restangular object. The object should have an ID if it has already been created.

"queryParams" the same as the `get` and `post` methods for adding extra query arguments (can be skipped).

Here is an example:

```
   var data = {id:3,title:"Something"};
   var document = Restangular.restangularizeElement(null, {id:3,title:"Something"}, 'documents')
```

This would create a `document` object that has an ID or `3` and `title` of "Something" and supports Restangular methods like `put` and `post`.
